lib/spack/spack/cmd/info.py

```diff
@@ -189,12 +189,11 @@ def print_text_info(pkg):
         # Here we sort first on the fact that a version is marked
         # as preferred in the package, then on the fact that the
         # version is not develop, then lexicographically
-        l = [
-            (value.get('preferred', False), not key.isdevelop(), key)
-            for key, value in pkg.versions.items()
-        ]
-        l = sorted(l)
-        _, _, preferred = l.pop()
+        preferred = sorted(
+            pkg.versions, key=lambda v: (
+                pkg.versions[v].get('preferred', False),
+                not v.isdevelop(),
+                v)).pop()
```

Comment:
Minor comment, but I find the previous version much more readable: the tuple is on a single line, sorting and popping are separate operations. If we want to rework it this way, maybe use:
```python
key_fn = lambda v: (pkg.versions[v].get('preferred', False), not v.isdevelop(), v)
preferred = sorted(pkg.versions, key=key_fn).pop()
```
?

Reply:
sure -- that's more readable!

Reply:
Personally I'm not such a fan of forbidding `l` as a variable name, but we already have exceptions to PEP8, and the point is to be as compliant as possible with Python OSS conventions, to make it easy to onboard people.  And in most cases I changed there were probably more descriptive names than `l` that could've been used so :man_shrugging:.

Reply:
Completely agree with your statement above.